Fix translation of Cinder storage classess to CSI

jsafrane opened the pull request:
/kind bug

**What this PR does / why we need it**:
In-tree Cinder storage class must be translated to CSI:

* `sc.parameters["fsType"]` -> `sc.parameters["csi.storage.k8s.io/fstype"]`
* ~`sc.parameters["availability"]` -> `sc.allowedTopology` (with CSI topology keys)~
* `sc.allowedTopology` (with in-tree topology keys) -> `sc.allowedTopology`  (with CSI topology keys)

Found by running e2e tests with CSI migration enabled. One topology test failed:

```
[sig-storage] In-tree Volumes [Driver: cinder] [Testpattern: Dynamic PV (delayed binding)] topology should provision a volume and schedule a pod with AllowedTopologies [Suite:openshift/conformance/parallel] [Suite:k8s]

...

ProvisioningFailed: failed to provision volume with StorageClass "e2e-topology-1939-cinder-scf2rsm": error generating accessibility requirements: topology map[topology.cinder.csi.openstack.org/zone:nova] from selected node "jsafrane-5x4gw-worker-0-vc2z9" is not in requisite: [map[failure-domain.beta.kubernetes.io/zone:nova]]
```

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Fixed provisioning of Cinder volumes migrated to CSI when StorageClass with AllowedTopologies was used.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

jsafrane committed:
Fix translation of Cinder storage classess to CSI

In-tree Cinder storage class must be translated to CSI:
- sc.params["fsType"] -> sc.params["csi.storage.k8s.io/fstype"]
- sc.allowedTopology (with in-tree topology keys) -> sc.allowedTopology
  (with CSI topology keys)

jsafrane commented:
/sig storage
/priority important-soon

jsafrane commented:
-retest

jsafrane commented:
/retest

ialidzhikov commented:
/test pull-kubernetes-integration

jsafrane commented:
cc @kubernetes/provider-openstack-pr-reviews, PTAL.

msau42 commented:
/approve
/triage accepted

Will let openstack folks give final review

ramineni commented:
@jsafrane sc.parameters["availability"] is still supported in CSI driver and takes priority over other topology keys if specified. So we dont need to translate the same?

Fedosin commented on staging/src/k8s.io/csi-translation-lib/plugins/openstack_cinder_test.go:
```diff
@@ -0,0 +1,80 @@
+		}
+
+		if !reflect.DeepEqual(got, tc.expSc) {
+			t.Errorf("Got parameters: %v, expected :%v", got, tc.expSc)
```
nit: typo 
```suggestion
			t.Errorf("Got parameters: %v, expected: %v", got, tc.expSc)
```

jsafrane commented:
@ramineni, you're right, I removed `availability` translation and added explicit unit tests it's not translated.

Still, migration to CSI was a great opportunity to drop the hack we had in Kubernetes StorageClasses and stop supporting two conflicting ways how to place volumes into zones.  I did not find a clear statement if the CSI driver is GA or beta or alpha, is it too late to remove `availability` support from there? (If it's at least beta, then the answer is yes, it's too late).

Fedosin commented:
/test pull-kubernetes-e2e-kind

Fedosin commented:
/test pull-kubernetes-e2e-kind-ipv6

Fedosin commented:
/test pull-kubernetes-verify

mandre commented:
> @ramineni, you're right, I removed `availability` translation and added explicit unit tests it's not translated.

Can you update the PR description and commit message to reflect the change you just made?

ramineni commented:
> @ramineni, you're right, I removed `availability` translation and added explicit unit tests it's not translated.
> 
> Still, migration to CSI was a great opportunity to drop the hack we had in Kubernetes StorageClasses and stop supporting two conflicting ways how to place volumes into zones. I did not find a clear statement if the CSI driver is GA or beta or alpha, is it too late to remove `availability` support from there? (If it's at least beta, then the answer is yes, it's too late).

@jsafrane we did consider once before, but there was some issue (something with volume AZ , node AZ can be different). Dont remember exactly, I suppose, I can do some digging again if we could remove the redundancy. Thanks

jsafrane commented:
Ok, this PR does not touch `availability` now. I updated the commit message with the last push. PTAL

ramineni commented:
Thanks.
/lgtm

jsafrane commented:
Fixed `gofmt`

Fedosin reviewed (approved):
/lgtm
